components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/saml/SAML2BearerGrantHandler.java

```diff
@@ -749,4 +753,25 @@ protected AuthenticatedUser buildLocalUser(OAuthTokenReqMessageContext tokReqMsg
         return authenticatedUser;
     }
 
+    /**
+     * This method is setting the username removing the domain name without checking whether the user is federated
+     * or not. This fix has done for support backward capability.
+     *
+     * @param tokReqMsgCtx Token request message context.
+     * @param assertion    SAML2 Assertion.
+     */
+    protected void createLegacyUser(OAuthTokenReqMessageContext tokReqMsgCtx, Assertion assertion)
+            throws IdentityOAuth2Exception {
+        String resourceOwnerUserName = assertion.getSubject().getNameID().getValue();
+
+        if (StringUtils.isBlank(resourceOwnerUserName)) {
```

Comment:
Isn't this checked before hand ? within the method which this is getting called ?